documentation-site/components/version-selector.js

```diff
@@ -19,48 +19,50 @@ import {Button, KIND} from 'baseui/button';
 import {version} from '../../package.json';
 import versions from '../../versions.json';
 
-// this is the only place needed to be updated if a new
-// major is cut
-const MAJOR_VERSIONS = [
-  {label: 'v9'},
-  {label: 'v8'},
-  {label: 'v7'},
-  {label: 'v6'},
-  {label: 'v5'},
-  {label: 'v4'},
-  {label: 'v3'},
-  {label: 'v2'},
-  {label: 'v1'},
-].map(version => {
+const majorVersions = Array.from(
+  versions.reduce((set, version) => {
+    return set.add(semver.major(version.tag_name));
+  }, new Set()),
+).map(version => ({
```

Comment:
i just learned the other day that the second argument to `Array.from` is a map function to run over the new array. not sure it makes a difference though.